src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -3719,7 +3738,7 @@ bool getParamLimitsOfNewGeosForTrim(const SketchObject* obj,
 
 void createArcsFromGeoWithLimits(const Part::GeomCurve* geo,
                                  const std::vector<std::pair<double, double>>& paramsOfNewGeos,
-                                 std::vector<std::unique_ptr<Part::Geometry>>& newGeos)
+                                 std::vector<Part::Geometry*>& newGeos)

```

Comment:
it looks a bit strange to me to replace `unique_ptr` with raw pointers. I'm probably missing something, could you share your thoughts?

Reply:
I think this was done since that's the way these are passed in some cases.

This could also be because I couldn't get `unique_ptr` to work that particular day. As I implied in the OP, this was meant to be in draft until these methods are moved out of `SketchObject.cpp`. So I might have taken some liberties.

That said, please review this as a finalized PR and I will fix it as necessary.

Reply:
Just checked. The problem is `newGeos` is passed on to `replaceGeometries`, and IIRC it was messing up its usage. I think I will leave this be for now, especially since this method itself may be changed/removed.